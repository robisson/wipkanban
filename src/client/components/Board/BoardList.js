// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";

type Props = {
  boards: Array<Object>
};

const BoardList = ({ boards }: Props) => {
  let boardsRendered = boards.map(board => (
    <MiniBoard key={board._id} {...board} />
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <IconButton color="default" aria-label="Add board">
              <AddIcon />
            </IconButton>
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={0}>
        <Grid item lg={6}>
          {boardsRendered}
        </Grid>
      </Grid>
    </div>
  );
};

export default BoardList;
