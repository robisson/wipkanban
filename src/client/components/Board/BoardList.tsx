// @flow
import * as React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import MiniBoard from "./MiniBoard";
import Tooltip from "@material-ui/core/Tooltip";
import AddBoard from "./AddBoardContainer";

interface IBoardListProps {
  boards: Array<Object>;
}

const BoardList: React.StatelessComponent<any> = ({
  boards
}: IBoardListProps) => {
  let boardsRendered = boards.map((board: any) => (
    <Grid key={board._id} item lg={6}>
      <MiniBoard key={board._id} {...board} />
    </Grid>
  ));

  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <DeveloperBoardIcon /> My boards
          <Tooltip title="Add team">
            <AddBoard />
          </Tooltip>
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="wrap" spacing={8}>
        {boardsRendered}
      </Grid>
    </div>
  );
};

export default BoardList;
