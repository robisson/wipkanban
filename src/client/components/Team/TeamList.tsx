import * as React from "react";
import Typography from "@material-ui/core/Typography";
import Tooltip from "@material-ui/core/Tooltip";
import AddIcon from "@material-ui/icons/AddCircle";
import IconButton from "@material-ui/core/IconButton";
import GroupIcon from "@material-ui/icons/Group";

const TeamList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="h6" gutterBottom>
          <GroupIcon /> My teams
        </Typography>
      </div>
      <div style={{ textAlign: "center" }}>
        <Typography component="p" gutterBottom>
          Create your first team
        </Typography>
        <Tooltip title="Add team">
          <IconButton color="default" aria-label="Add team">
            <AddIcon />
          </IconButton>
        </Tooltip>
      </div>
    </div>
  );
};

export default TeamList;
