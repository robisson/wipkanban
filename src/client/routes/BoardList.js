// @flow
import * as React from "react";
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import DeveloperBoardIcon from '@material-ui/icons/DeveloperBoard';
import MiniBoard from './MiniBoard';

const BoardList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="title" gutterBottom>
          <DeveloperBoardIcon/>{' '}
          My boards
        </Typography>
      </div>
      <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
        <Grid item lg={6}>
          <MiniBoard/>
        </Grid>
      </Grid>
    </div>
  )
};

export default BoardList;
