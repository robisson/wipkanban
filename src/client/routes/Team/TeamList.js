// @flow
import * as React from "react";
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import AddIcon from '@material-ui/icons/AddCircle';
import IconButton from '@material-ui/core/IconButton';
import GroupIcon from '@material-ui/icons/Group';

const TeamList = () => {
  return (
    <div>
      <div>
        <Typography color="secondary" variant="title" gutterBottom>
          <GroupIcon/>{' '}
          My teams
        </Typography>
      </div>
      <center>
        <Typography component="p" gutterBottom>
          Create your first team
        </Typography>
        <Tooltip title="Add team">
          <IconButton color="default" aria-label="Add team">
            <AddIcon/>
          </IconButton>
        </Tooltip>
      </center>
    </div>
  )
};

export default TeamList;
