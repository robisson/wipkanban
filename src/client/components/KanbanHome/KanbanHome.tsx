import * as React from "react";
import Grid from '@material-ui/core/Grid';
import BoardList from '../Board/BoardListContainer';
import TeamList from '../Team/TeamList';

const KanbanHome = () => {
  return (
    <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
      <Grid item lg={6}>
        <BoardList/>
      </Grid>
      <Grid item lg={6}>
        <TeamList/>
      </Grid>
    </Grid>
  )
};

export default KanbanHome;
