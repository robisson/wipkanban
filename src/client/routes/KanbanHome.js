// @flow
import * as React from "react";
import Grid from '@material-ui/core/Grid';
import BoardList from './BoardList';
import TeamList from './TeamList';

const KanbanHome = () => {
  return (
    <Grid container justify="flex-start" wrap="nowrap" spacing={0}>
      <Grid item lg={6}>
        <BoardList/>
      </Grid>
      <Grid item lg={6}>
        <TeamList/>
      </Grid>
    </Grid>
  )
};

export default KanbanHome;
