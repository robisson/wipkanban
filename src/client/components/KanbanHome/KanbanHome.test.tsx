import React from "react";
import KanbanHome from "./index";
import { createShallow } from "@material-ui/core/test-utils";
import Grid from "@material-ui/core/Grid";
import BoardList from "../Board/BoardListContainer";
import TeamList from "../Team/TeamList";

describe("KanbanHome component", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("should render 1 containing 1 TeamList and BoardListComponent", () => {
    let wrapper = shallow(<KanbanHome />);
    expect(wrapper.type()).toEqual(Grid);
    expect(wrapper.find(BoardList).type()).toEqual(BoardList);
    expect(wrapper.find(TeamList).type()).toEqual(TeamList);
    expect(wrapper).toMatchSnapshot();
  });
});
