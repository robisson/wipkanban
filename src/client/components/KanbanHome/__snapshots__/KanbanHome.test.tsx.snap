// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KanbanHome component should render 1 containing 1 TeamList and BoardListComponent 1`] = `
<WithStyles(Grid)
  container={true}
  justify="flex-start"
  spacing={0}
  wrap="nowrap"
>
  <WithStyles(Grid)
    item={true}
    lg={6}
  >
    <Connect(BoardListContainer) />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    lg={6}
  >
    <TeamList />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
